### PR TITLE
Custom GET Endpoint

### PR DESCRIPTION
**Pull request by neilcook**
- Custom GET Endpoint (see #261)
- New Lua functions to retrieve the various black/whitelists in their entirety
- Use %{python3_version} in the report_api config rather than hardcode it

### CODE REVIEW

**Comment by sshipway** [comment]
The Get endpoint combined with the function to retrieve black/whitelists will solved #261

File: common/wforce-webserver.cc [hunk 1]
```diff
@@ -71,7 +71,7 @@ size_t WforceWebserver::getNumConns()
   return d_sock_vec.size();
 }
 
-bool WforceWebserver::registerFunc(const std::string& command, HTTPVerb verb, WforceWSFunc func)
+bool WforceWebserver::registerFunc(const std::string& command, HTTPVerb verb, const WforceWSFunc& func)
```
Comment (cmouse) [inline common/wforce-webserver.cc]: unrelated change.

File: common/wforce-webserver.cc [hunk 2]
```diff
@@ -224,17 +224,21 @@ void WforceWebserver::connectionThread(WforceWebserver* wws)
         // set the defaults in case we don't find a command
         resp.status = 404;
         resp.body = R"({"status":"failure", "reason":"Command not found"})";
-    
+
+        std::string new_content_type;
```
Comment (cmouse) [inline common/wforce-webserver.cc]: totally unrelated change.

Comment (neilcook) [reply common/wforce-webserver.cc]: Why is this unrelated? The purpose of this new function is to be able to return different content types from the GET functions. Not also allowing that for the other types of functions would be reverse

File: common/wforce-webserver.cc (hunk 1 above)
Comment (neilcook) [reply common/wforce-webserver.cc]: will break out into a single commit

File: common/wforce-webserver.cc (hunk 2 above)
Comment (cmouse) [reply common/wforce-webserver.cc]: i cannot see how it's related to the commit, so perhaps you should do this as a preparation before this commit?

Comment (neilcook) [reply common/wforce-webserver.cc]: ok
